test/tally_tests.erl

```diff
@@ -311,7 +333,8 @@ tally_issue_14_test() ->
 % }]
 tally_foo2_test() ->
   % changing variable order might produce a different number of solutions
```

Comment:
Added a test showcasing why variable order matters.
It's hinted at in `Polymorphic Functions with Set-Theoretic Types Part 2` in the Appendix:

> As stated in Section C.1, we chose an arbitrary ordering on type variables, which affects the generated
substitutions and then the resulting types.

For the constraint

$(\alpha_2, 42) \subseteq (\alpha_1, 42)$

the variable ordering specifies whether $a_1$ is an upper or lower bound for $a_2$.
The tally solution essentially generates two solutions based on that reasoning:

* If $\alpha_2 < \alpha_1$, then $\alpha_2$ is replaced by $\alpha_1 \cap \mu2$
  * the constraint $(\alpha_1 \cap \mu_2, 42) \subseteq (\alpha_1 \cap \mu_2, 42)$ holds for all possible substitutions

* If $\alpha_1 < \alpha_2$, then either
  * $\alpha_2$ is replaced by $\emptyset$
    * the constraint $(\emptyset, 42) \subseteq (\alpha_1, 42)$ holds for all possible substitutions 
  *  $\alpha_1$ is replaced by $\alpha_2 \cup \mu_1$
     * the constraint $(\alpha_2, 42) \subseteq (\alpha_2 \cup \mu_1, 42)$ holds for all possible substitutions

Via the tally completeness theorem both solutions can be transformed into each other, thus making them equivalent.

It's therefore required from now on that each tally test specifies the variable ordering, otherwise the solutions might change at any point in the future.